grafana/grafana
# Share azureauth between prometheus clients

### Description
We have azureauth logic in buffered folder (client) but it was shared querydata client as well. 
I refactored this and make the logic a bit more accessible. 

Fixes: https://github.com/grafana/observability-metrics-squad/issues/52

### How can you test if this is working?
- In your `custom.ini` enable azure auth 
```
[auth]
azure_auth_enabled=true
```
- Launch Grafana via VSCode so you can put breakpoints.
- Put a breakpoint via VSCode in `<Your go home dir>pkg/mod/github.com/grafana/grafana-azure-sdk-go@v1.3.1/azhttpclient/middleware.go`
- Create a new prometheus datasource
- Enable Azure Authentication
- Use the credentials from azuremonitoring (ask me for details if you couldn't find them)
- Hit Save&Test and wait for breakpoint hit and check you have a token
